src/gmastertrack.cxx

```diff
@@ -170,11 +170,16 @@ static void gmastertrack_button_callback(Fl_Widget *w, void *data)
 
 	} else if ( strcmp( w->label(), "Tap" ) == 0 ) {
 		if ( Fl::event_button() == FL_RIGHT_MOUSE ) {
-			const char* answer = fl_input("Enter BPM value: ");
+			string question = "Enter BPM value (between " + to_string(MIN_TEMPO) + " and " 
+								+ to_string(MAX_TEMPO) + "):";
+			const char* answer = fl_input(question.c_str());
```

Comment:
This line gives a compiler warning, it has to do with the formatting of the string - I'm not sure exactly where it comes from. Please investigate, and if you can't reproduce let me know and I'll dig into it:
```
../src/gmastertrack.cxx:175:50: warning: format not a string literal and no format arguments [-Wformat-security]
    const char* answer = fl_input(question.c_str());
```

Reply:
Dont have this warning:

```
-- Architecture:  x86_64
-- Configuring done
-- Generating done
-- Build files have been written to: /home/georg/workspace/openAV-Luppp/build
Scanning dependencies of target luppp
[  2%] Building CXX object src/CMakeFiles/luppp.dir/controllerupdater.cxx.o
[  4%] Building CXX object src/CMakeFiles/luppp.dir/audiobuffer.cxx.o
[  7%] Building CXX object src/CMakeFiles/luppp.dir/debug.cxx.o
[  9%] Building CXX object src/CMakeFiles/luppp.dir/diskreader.cxx.o
[ 12%] Building CXX object src/CMakeFiles/luppp.dir/diskwriter.cxx.o
[ 14%] Building CXX object src/CMakeFiles/luppp.dir/event.cxx.o
[ 17%] Building CXX object src/CMakeFiles/luppp.dir/eventhandlerdsp.cxx.o
[ 19%] Building CXX object src/CMakeFiles/luppp.dir/eventhandlergui.cxx.o
[ 21%] Building CXX object src/CMakeFiles/luppp.dir/gaudioeditor.cxx.o
[ 24%] Building CXX object src/CMakeFiles/luppp.dir/gmastertrack.cxx.o
[ 26%] Building CXX object src/CMakeFiles/luppp.dir/goptions.cxx.o
[ 29%] Building CXX object src/CMakeFiles/luppp.dir/gridlogic.cxx.o
[ 31%] Building CXX object src/CMakeFiles/luppp.dir/gtrack.cxx.o
[ 34%] Building CXX object src/CMakeFiles/luppp.dir/gui.cxx.o
In file included from /home/georg/workspace/openAV-Luppp/src/gtrack.cxx:19:0:
/home/georg/workspace/openAV-Luppp/src/gtrack.hxx: In Konstruktor »GTrack::GTrack(int, int, int, int, const char*)«:
/home/georg/workspace/openAV-Luppp/src/gtrack.hxx:105:20: Warnung: »GTrack::jackSendActivate« wird initialisiert nach [-Wreorder]
  Avtk::LightButton jackSendActivate;
                    ^~~~~~~~~~~~~~~~
/home/georg/workspace/openAV-Luppp/src/gtrack.hxx:95:15: Warnung:   »Avtk::Box GTrack::sndBox« [-Wreorder]
  Avtk::Box    sndBox;
               ^~~~~~
/home/georg/workspace/openAV-Luppp/src/gtrack.cxx:36:1: Warnung:   während es hier initialisiert wurde [-Wreorder]
 GTrack::GTrack(int x, int y, int w, int h, const char* l ) :
 ^~~~~~
In file included from /home/georg/workspace/openAV-Luppp/src/gtrack.cxx:19:0:
/home/georg/workspace/openAV-Luppp/src/gtrack.hxx:115:20: Warnung: »GTrack::panDial« wird initialisiert nach [-Wreorder]
  Avtk::Dial        panDial;
                    ^~~~~~~
/home/georg/workspace/openAV-Luppp/src/gtrack.hxx:98:15: Warnung:   »Avtk::Volume GTrack::volume« [-Wreorder]
  Avtk::Volume volume;
               ^~~~~~
/home/georg/workspace/openAV-Luppp/src/gtrack.cxx:36:1: Warnung:   während es hier initialisiert wurde [-Wreorder]
 GTrack::GTrack(int x, int y, int w, int h, const char* l ) :
 ^~~~~~
[ 36%] Building CXX object src/CMakeFiles/luppp.dir/jack.cxx.o
In file included from /home/georg/workspace/openAV-Luppp/src/gui.cxx:19:0:
/home/georg/workspace/openAV-Luppp/src/gui.hxx: In Konstruktor »Gui::Gui(const char*)«:
/home/georg/workspace/openAV-Luppp/src/gui.hxx:126:19: Warnung: »Gui::window« wird initialisiert nach [-Wreorder]
  Fl_Double_Window window;
                   ^~~~~~
/home/georg/workspace/openAV-Luppp/src/gui.hxx:99:7: Warnung:   »bool Gui::enablePerTrackOutput« [-Wreorder]
  bool enablePerTrackOutput;
       ^~~~~~~~~~~~~~~~~~~~
/home/georg/workspace/openAV-Luppp/src/gui.cxx:360:1: Warnung:   während es hier initialisiert wurde [-Wreorder]
 Gui::Gui(const char* argZero) :
 ^~~
/home/georg/workspace/openAV-Luppp/src/jack.cxx: In Elementfunktion »void Jack::processFrames(int)«:
/home/georg/workspace/openAV-Luppp/src/jack.cxx:531:28: Warnung: Vergleich zwischen vorzeichenbehafteten und vorzeichenlosen Ganzzahlausdrücken [-Wsign-compare]
  for(unsigned int i = 0; i < nframes; i++) {
                          ~~^~~~~~~~~
/home/georg/workspace/openAV-Luppp/src/jack.cxx:610:24: Warnung: Vergleich zwischen vorzeichenbehafteten und vorzeichenlosen Ganzzahlausdrücken [-Wsign-compare]
  if(lastnframes+nframes<buffers.nframes) {
     ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
[ 39%] Building CXX object src/CMakeFiles/luppp.dir/jacksendreturn.cxx.o
[ 41%] Building CXX object src/CMakeFiles/luppp.dir/logic.cxx.o
In file included from /home/georg/workspace/openAV-Luppp/src/jacksendreturn.cxx:1:0:
/home/georg/workspace/openAV-Luppp/src/jacksendreturn.hxx: In Konstruktor »JackSendReturn::JackSendReturn(int, AudioProcessor*, jack_client_t*)«:
/home/georg/workspace/openAV-Luppp/src/jacksendreturn.hxx:53:18: Warnung: »JackSendReturn::m_previousProcessor« wird initialisiert nach [-Wreorder]
  AudioProcessor* m_previousProcessor;
                  ^~~~~~~~~~~~~~~~~~~
/home/georg/workspace/openAV-Luppp/src/jacksendreturn.hxx:47:8: Warnung:   »float JackSendReturn::m_sendvol« [-Wreorder]
  float m_sendvol;
        ^~~~~~~~~
/home/georg/workspace/openAV-Luppp/src/jacksendreturn.cxx:6:1: Warnung:   während es hier initialisiert wurde [-Wreorder]
 JackSendReturn::JackSendReturn(int trackid, AudioProcessor *prev, jack_client_t *client)
 ^~~~~~~~~~~~~~
/home/georg/workspace/openAV-Luppp/src/jacksendreturn.cxx: In Elementfunktion »virtual void JackSendReturn::process(unsigned int, Buffers*)«:
/home/georg/workspace/openAV-Luppp/src/jacksendreturn.cxx:52:16: Warnung: Vergleich zwischen vorzeichenbehafteten und vorzeichenlosen Ganzzahlausdrücken [-Wsign-compare]
  for(int i=0; i<nframes; i++) {
               ~^~~~~~~~
[ 43%] Building CXX object src/CMakeFiles/luppp.dir/looper.cxx.o
[ 46%] Building CXX object src/CMakeFiles/luppp.dir/looperclip.cxx.o
[ 48%] Building CXX object src/CMakeFiles/luppp.dir/main.cxx.o
[ 51%] Building CXX object src/CMakeFiles/luppp.dir/metronome.cxx.o
[ 53%] Building CXX object src/CMakeFiles/luppp.dir/timemanager.cxx.o
[ 56%] Building CXX object src/CMakeFiles/luppp.dir/trackoutput.cxx.o
/home/georg/workspace/openAV-Luppp/src/timemanager.cxx: In Elementfunktion »void TimeManager::setTransportState(TRANSPORT_STATE)«:
/home/georg/workspace/openAV-Luppp/src/timemanager.cxx:173:17: Warnung: Vergleich zwischen vorzeichenbehafteten und vorzeichenlosen Ganzzahlausdrücken [-Wsign-compare]
   for(int i=0; i<observers.size(); i++)
                ~^~~~~~~~~~~~~~~~~
[ 58%] Building CXX object src/CMakeFiles/luppp.dir/avtk/bindings.cxx.o
[ 60%] Building CXX object src/CMakeFiles/luppp.dir/avtk/clipselector.cxx.o
[ 63%] Building CXX object src/CMakeFiles/luppp.dir/controller/controller.cxx.o
[ 65%] Building CXX object src/CMakeFiles/luppp.dir/controller/genericmidi.cxx.o
[ 68%] Building CXX object src/CMakeFiles/luppp.dir/controller/guicontroller.cxx.o
[ 70%] Building CXX object src/CMakeFiles/luppp.dir/controller/nonseq.cxx.o
[ 73%] Building CXX object src/CMakeFiles/luppp.dir/observer/midi.cxx.o
[ 75%] Building CXX object src/CMakeFiles/luppp.dir/observer/time.cxx.o
[ 78%] Building CXX object src/CMakeFiles/luppp.dir/state/state.cxx.o
[ 80%] Building CXX object src/CMakeFiles/luppp.dir/state/stately.cxx.o
[ 82%] Building CXX object src/CMakeFiles/luppp.dir/tests/diskreadertest.cxx.o
[ 85%] Building CXX object src/CMakeFiles/luppp.dir/tests/diskwritertest.cxx.o
[ 87%] Building CXX object src/CMakeFiles/luppp.dir/tests/goptionstest.cxx.o
[ 90%] Building CXX object src/CMakeFiles/luppp.dir/tests/gridlogictests.cxx.o
[ 92%] Linking CXX executable luppp
[100%] Built target luppp
 ~/workspace/openAV-Luppp/build   sanitycheck ●  ./compile.sh
-- Architecture:  x86_64
-- Configuring done
-- Generating done
-- Build files have been written to: /home/georg/workspace/openAV-Luppp/build
Scanning dependencies of target luppp
[  2%] Building CXX object src/CMakeFiles/luppp.dir/main.cxx.o
[  4%] Linking CXX executable luppp
[100%] Built target luppp
```